sdk/search/Azure.Search.Documents/src/AzureSearchDocumentsEventSource.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.Tracing;
+using Azure.Core.Diagnostics;
+
+namespace Azure.Search.Documents
+{
+    [EventSource(Name = EventSourceName)]
+    internal sealed class AzureSearchDocumentsEventSource : EventSource
+    {
+        private const string EventSourceName = "Azure-Search-Documents";
+
+        internal const int BatchActionCountUpdatedEvent = 1;
+        internal const int BatchSubmittedEvent = 2;
+        internal const int RetryQueueResizedEvent = 3;
+        internal const int PendingQueueResizedEvent = 4;
+
+        private AzureSearchDocumentsEventSource() :
+            base(EventSourceName, EventSourceSettings.Default, AzureEventSourceListener.TraitName, AzureEventSourceListener.TraitValue)
+        { }
+
+        public static AzureSearchDocumentsEventSource Instance { get; } = new AzureSearchDocumentsEventSource();
+
+        [Event(PendingQueueResizedEvent, Level = EventLevel.Verbose, Message = "Indexing publisher pending queue size = {0}.")]
+        public void PendingQueueResized(int queueSize)
```

Comment:
These should all say `SearchIndexingBufferedSender` because there are other types of indexing that they don't apply to and there's no public type called Publisher.